src/engine/threaded_engine.h

```diff
@@ -384,6 +384,10 @@ class ThreadedEngine : public Engine {
           threaded_opr->opr_exception =
               std::make_shared<std::exception_ptr>(std::current_exception());
           callback();
+        } catch (std::exception& e) {
```

Comment:
Can we just replace the `catch (dmlc::Error& e)` on L383 with `catch (const std::exception& e)` here?